### DISCUSSION TITLE
Does micronucleus work on the attiny202?

### DISCUSSION THREAD

**Discussion by KenwoodFox** [micronucleus/micronucleus#269] [Q&A] [answered]
Hey micronucleus! I was wondering, because im having a hard time sourcing attiny85's.. if micronucleus works on the attiny202 witch is much more available i think!

Or maybe there are similar chips? i want to include an attiny on a board im working on, but without external uart or specialized programmer, thanks for reading!

**Comment by cpldcpu** [comment, marked as answer]
In principle it should be possible to port Micronucleus also to newer ATtiny. But this has not been done yet.

On ATtiny202 you can also use UPDI, which just requires a single wire and a USB2TTL interface:
https://github.com/microchip-pic-avr-tools/pymcuprog

**Reply by KenwoodFox** [reply]
That still requires a sperate usb ttl chip, witch is what im trying to avoid. My current design is very nice in it only requires a single chip. 
![image](https://user-images.githubusercontent.com/31870999/193475017-e2790c94-9e97-4cbd-a69e-44293d26a346.png)

Maybe i can help experiment porting to the 202? where would i start? or is there a better place to source 85's? they're really expensive/sold out everywhere i checked :c

**Reply by cpldcpu** [reply]
Did you manage to get V-USB to work on ATtiny202? Btw, the 202 only has 2kb of flash. Micronucleus would eat up most of that.

Digikey still seems to have plenty of ATtiny85, btw.

**Reply by KenwoodFox** [reply]
Ugh they're back in stock, thank you and sorry!
